src/tcache.c

```diff
@@ -679,19 +677,19 @@ tcache_init(tsd_t *tsd, tcache_slow_t *tcache_slow, tcache_t *tcache,
 
 	/*
 	 * We reserve cache bins for all small size classes, even if some may
-	 * not get used (i.e. bins higher than tcache_nhbins).  This allows
+	 * not get used (i.e. bins higher than tcache_nbins).  This allows
 	 * the fast and common paths to access cache bin metadata safely w/o
 	 * worrying about which ones are disabled.
 	 */
-	unsigned tcache_nhbins = tcache_nhbins_get(tcache);
-	unsigned n_reserved_bins = tcache_nhbins < SC_NBINS ? SC_NBINS
-	    : tcache_nhbins;
+	unsigned tcache_nbins = tcache_nbins_get(tcache_slow);
+	unsigned n_reserved_bins = tcache_nbins < SC_NBINS ? SC_NBINS
+	    : tcache_nbins;
```

Comment:
The whole `n_reserved_bins` isn't necessary anymore, is that right? If so let's get rid of this section (including the memset below).